Solution1/ClassLibrary1/NumericUpDownLight.cs

```diff
@@ -0,0 +1,249 @@
+﻿using System;
+using System.ComponentModel;
+using System.Drawing;
+using System.Windows.Forms;
+
+namespace ClassLibrary1
+{
+    public class NumericUpDownLight : Control
+    {
+        private int _value = 0;
+        private EventHandler _valueChanged = null;
+        private HorizontalAlignment _textAlign = HorizontalAlignment.Left;
+        private Color _mouseHoverBackColor = Color.Blue;
+
+        private readonly CustomTextBox TextBox = new CustomTextBox();
+        private readonly CustomButton ButtonUp = new CustomButton();
+        private readonly CustomButton ButtonDown = new CustomButton();
+
+        private bool IsBackgroundChanged = false;
+
+        [Browsable(false)]
+        public override string Text { get; set; }
+
+        [
+        Category("Data"),
+        Description("Specifies the value of the control.")
+        ]
+        public int Value
+        {
+            get
+            {
+                return _value;
+            }
+            set
+            {
+                _value = value;
+                base.Invalidate();
+            }
+        }
+
+        [
+        Category("Property Changed"),
+        Description("ValueChanged event triggers after the value was changed")
+        ]
+        public event EventHandler ValueChanged
```

Comment:
Net has syntactic sugar for events.
public event EventHandler ValueChanged; - without private field;